manifests/0000_90_kube-controller-manager-operator_03_servicemonitor.yaml

```diff
@@ -42,3 +42,21 @@ spec:
       for: 15m
       labels:
         severity: critical
+---
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: kube-controller-manager-operator
+  namespace: openshift-kube-controller-manager-operator
+spec:
+  groups:
+  - name: cluster-version
+    rules:
+    - alert: KubeControllerManagerDown
+      annotations:
+        message: KubeControllerManager has disappeared from Prometheus target discovery.
+      expr: |
+        kube_poddisruptionbudget_status_expected_pods <= on(poddisruptionbudget, service) kube_poddisruptionbudget_status_desired_healthy
```

Comment:
This should print the PDB and its service when expected pods will less or equal than desired healthy number.

Reply:
> ...less or equal than desired healthy number...

Is this what we want?  We don't want to alert when you drop <= `minAvailable` or some such?  Does this fire during upgrade when we cycle a control plane machine and at least while that is happening should be one less than our desired count but still not violating our disruption budget.

Reply:
Pointed [PromeCIeus][1] at [the CI upgrade run][2], looks like this is fine:

```
(max(kube_poddisruptionbudget_status_expected_pods) by (poddisruptionbudget, service)) - (min (kube_poddisruptionbudget_status_desired_healthy) by (poddisruptionbudget, service))
```

![Screenshot from 2019-11-06 12-28-04](https://user-images.githubusercontent.com/209920/68335228-ef66be00-0090-11ea-86df-50885b73df76.png)

For example, `kube_poddisruptionbudget_status_expected_pods` is 3 for etcd-quorum-guard over that whole time range, while `kube_poddisruptionbudget_status_desired_healthy` is 2.

[1]: https://promecieus.svc.ci.openshift.org/
[2]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-kube-controller-manager-operator/301/pull-ci-openshift-cluster-kube-controller-manager-operator-master-e2e-aws-upgrade/209